python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -78,7 +88,8 @@ def _graph_data_to_fused_csc_sampling_graph(
         src, dst = read_edges(dataset_dir, edge_fmt, edge_path)
         num_nodes = graph_data["nodes"][0]["num"]
         num_edges = len(src)
-        coo_tensor = torch.tensor(np.array([src, dst]))
+        node_dtype = torch.int32 if num_nodes <= INT32_MAX else torch.int64
```

Comment:
```suggestion
        node_dtype = torch.int32 if num_nodes - 1 <= INT32_MAX else torch.int64
```
Because max node id is `num_nodes - 1`.

Reply:
Sorry, due to #7147 limitations, we need to take back these -1 changes. The length also needs to fit into the dtype in that PR.